### PR TITLE
[Bug fixed]: New users added through compare are not able to generate resume #352 

### LINKED ISSUE #352: [Bug]: New users added through compare are not able to generate resume
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When a non-existing user is added to the database using the compare feature, and later visits the landing page to generate their resume by entering their GitHub username, they will encounter an internal server error.

### Expected behavior

Non exisiting users added through compare page can be able to generate the resume from landing page as usual.

### Add ScreenShots

https://github.com/ashutosh-rath02/git-re/assets/65452005/4c98ca30-cec2-43d3-9018-2f98eb90775a

### On which device are you experiencing this bug?

_No response_

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by dhairyathedev**
Fixes #352

### CODE REVIEW

**Comment by ashutosh-rath02** [comment]
Thank you for submitting your pull request! 🙌 We'll review it as soon as possible. In the meantime, please ensure that your changes align with our [CONTRIBUTING.md](https://github.com/ashutosh-rath02/git-re/blob/main/CONTRIBUTING.md). If there are any specific instructions or feedback regarding your PR, we'll provide them here. Thanks again for your contribution! 😊

File: app/api/users/route.ts [hunk 1]
```diff
@@ -81,18 +73,19 @@ export async function POST(request: NextRequest) {
...
   } catch (error) {
-    console.error("Error updating username:", error);
-    return new Response("Internal Server Error", { status: 500 });
+    console.error("Error:", error);
+
+    if (error instanceof Error) {
+      return new Response(error.message || "Internal Server Error", { status: 500 });
+    }
+
+    return new Response("Unexpected Error", { status: 500 });
```
Comment (ashutosh-rath02) [inline app/api/users/route.ts:88]: Why are both errors having same code?

**Review by ashutosh-rath02 — CHANGES_REQUESTED** [review]

File: app/api/users/route.ts (hunk 1 above)
Comment (dhairyathedev) [reply app/api/users/route.ts:88]: Both of these issues result in an Internal Server Error, which is indicated by the 500 status code.

**Review by ashutosh-rath02 — COMMENTED** [review]
LGTM

**Pull request merged**